modules/azure-firewall/inbound.tf

```
@@ -0,0 +1,43 @@
+data "azurerm_public_ip" "pip" {
+  count               = length(var.firewall)
+  name                = azurerm_public_ip.main[count.index].name
+  resource_group_name = azurerm_public_ip.main[count.index].resource_group_name
+}
+
+resource "azurerm_firewall_nat_rule_collection" "main" {
+  count               = length(var.firewall)
+  name                = "dnatRule-${lookup(element(var.firewall, count.index), "name")}"
```

is this dnat rule prefix needed?
```suggestion
  name                = "dnat-${lookup(element(var.firewall, count.index), "name")}"
```

```suggestion
  name                = "${lookup(element(var.firewall, count.index), "name")}"
```